src/pages/User/Settings/LearningSettings/LearningSettings.tsx

```diff
@@ -21,8 +20,13 @@ const LearningSettings: React.FC = () => {
     onSubmit,
     fields,
     control,
+    fetchUserSettings,
   } = useLearningSettings();
 
+  useEffect(() => {
+    fetchUserSettings();
+  }, []);
+
```

Comment:
nie ma sensu wyciągać tego do komponentu `[LearningSettings.tsx]`. Po to mamy całą logikę w hooku żeby również useEffecty wywoływać w hooku. Więc ten useEffect możesz przenieść do `useLearningSettings` i nie musisz wtedy w ogole tutaj tutaj wyciągać `fetchUserSettings`